docker/cassandra/default-config/cassandra-env.sh.d/002-cassandra-limits.sh

```diff
@@ -1 +0,0 @@
-ulimit -l unlimited
```

Comment:
Running `ulimit` doesn't seem to be possible in k8s: https://github.com/kubernetes/kubernetes/issues/3595

Do we really need to run `ulimit -l unlimited` inside the container? AFAIK this is only relevant when swap is enabled, which doesn't seem to be very common in k8s clusters.

In case this *is* necessary for a production-ready C* deployment, following [this](https://github.com/instaclustr/cassandra-operator/issues/208#issuecomment-516029789) approach I would consider omitting the command from the container image and instead recommend a host-level tweak in the docs, for example disabling swap on all worker nodes if C* is intended to serve production traffic.

Reply:
@johananl you’re probably right that `ulimit -l unlimited` isn’t necessary.

Some background: C* wants to call `mlockall` on startup to prevent the JVM heap (which gets sized so the min = max — it never grows) from ever being paged out. This is done to prevent GC pause times from sky-rocketing, as any GC that does a full heap pass would cause everything to be paged back in.

That is really only a concern AFAIK if swap is enabled.

If C* can’t call `mlockall`, it will just print a warning and move on.